parsers/ENTSOE.py

```diff
@@ -493,13 +493,13 @@ def query_ENTSOE(
             parser="ENTSOE.py",
             message="target_datetime has to be a datetime in query_entsoe",
         )
-
-    params["periodStart"] = (target_datetime + timedelta(hours=span[0])).strftime(
-        "%Y%m%d%H00"  # YYYYMMDDHH00
-    )
-    params["periodEnd"] = (target_datetime + timedelta(hours=span[1])).strftime(
-        "%Y%m%d%H00"  # YYYYMMDDHH00
-    )
+    if span is not None:
```

Comment:
This is fine but then we should also make sure the `periodStart` and `period end` parameters are already in the parameters object and throw an exception if they are not.

Otherwise we might accidentally query entso-e without valid parameters in the future.

Also didn't this already overwrite the parameters that was sent from the query function?

Btw have you tried requesting all the capacity data from 2015 and forwards? Might be possible to get all the capacity data in one go for each region which would make backfills and retroactive updates trivial.